### PR TITLE
Fixed raw autocomplete field reset value issue

### PR DESCRIPTION
**Pull request by vinothvino42**
In this PR, I fixed the raw autocomplete dropdown issue #408

### CODE REVIEW

File: lib/widget/input/dropdown.dart [hunk 1]
```diff
@@ -179,14 +185,28 @@ class SelectOneController extends FormFieldController {
...
 
 class SelectOneState extends FormFieldWidgetState<SelectOne> {
   final focusNode = FocusNode();
+  late TextEditingController rawTextEditingController;
+  late StreamSubscription<dynamic> streamSubscription;
+
   @override
   void initState() {
+    rawTextEditingController = TextEditingController();
+    streamSubscription =
```
Comment (vusters) [inline lib/widget/input/dropdown.dart:202]: This pattern is OK but in Controller we already have a way to invoke the widgetState's setState(). Just use `notifyListeners()`.

File: lib/widget/input/dropdown.dart [hunk 2]
```diff
@@ -179,14 +185,28 @@ class SelectOneController extends FormFieldController {
...
   int gap = 0;
   bool autoComplete = false;
+  final StreamController<dynamic> fieldController = StreamController<dynamic>();
 
   framework.EnsembleAction? onChange;
 }
 
 class SelectOneState extends FormFieldWidgetState<SelectOne> {
   final focusNode = FocusNode();
+  late TextEditingController rawTextEditingController;
```
Comment (vusters) [inline lib/widget/input/dropdown.dart:196]: just create the instance here and remove late.

File: lib/widget/input/dropdown.dart [hunk 3]
```diff
@@ -99,6 +98,13 @@ abstract class SelectOne extends StatefulWidget
     return null;
   }
 
+  void _setValue(dynamic value) {
+    _controller.maybeValue = value;
+    if (value is String && value == '') {
```
Comment (vusters) [inline lib/widget/input/dropdown.dart]: See my comment on the Github issue. <Empty string> maybe an actual value users want to use, so we reset only on `null`, plus expose a clear() method for the user to clear out the value.

File: lib/widget/input/dropdown.dart [hunk 4]
```diff
@@ -99,6 +98,13 @@ abstract class SelectOne extends StatefulWidget
     return null;
   }
 
+  void _setValue(dynamic value) {
+    _controller.maybeValue = value;
```
Comment (vusters) [inline lib/widget/input/dropdown.dart:102]: I know this is the existing code but it maybe better to check the validity here and either ignore to set the value (vs waiting for later). No more maybeValue.

File: lib/widget/input/dropdown.dart [hunk 5]
```diff
@@ -61,7 +60,7 @@ abstract class SelectOne extends StatefulWidget
   @override
   Map<String, Function> setters() {
     return {
-      'value': (value) => _controller.maybeValue = value,
+      'value': (value) => _setValue(value),
```
Comment (vusters) [inline lib/widget/input/dropdown.dart:63]: if you put this method on the controller, you can force the widgetState to call setState() using `notifyListeners()` inside Controller

**Review by vusters — APPROVED** [review]

**Comment by vinothvino42** [comment]
@vusters @kmahmood74 
Added clear method and you can set null to a value property of dropdown to reset the value.

**Sample YAML**:
```yaml
View:
  header:
    title: Reset Dropdown
  Column:
    styles:
      gap: 24
      padding: 24

    children:
      - Markdown:
          text: |
            ## Reset Dropdown value example

      - Dropdown:
          id: dropDown
          label: 1 - array of strings
          hintText: Select something
          items:
            - Red
            - Blue
            - Green

      - Text:
          id: onChangeDemo
          text: ${dropDown.value}

      - Button:
          label: Reset
          onTap: |
            //@code
            dropDown.clear();
            //dropDown.value = null;

```

**Pull request merged**